### PR TITLE
feat(packages): update tiup ctl package config for range [v8.4.0, )

### PR DESCRIPTION
**Pull request by wuhuizuo**
- support zip type url, oci artifacts, or local file.
- update etcd url for ctl pkg in range [v8.4.0, )

Signed-off-by: wuhuizuo <wuhuizuo@126.com>

### CODE REVIEW

File: packages/packages.yaml.tmpl [hunk 1]
```diff
@@ -48,16 +48,73 @@ components:
...
-      - description: interpret versions according to semantic version spec.
-        if: {{ semver.CheckConstraint ">= 6.1.0-0" .Release.version }}
+      - description: For range [v8.4.0, )
+        if: {{ semver.CheckConstraint ">= 8.4.0-0" .Release.version }}
         os: [linux, darwin]
         arch: [amd64, arm64]
         profile: [release, experiment]
+        artifacts:
+          - name: "ctl-{{ .Release.version }}-{{ .Release.os }}-{{ .Release.arch }}.tar.gz"
+            files:
```
Comment (wuhuizuo) [inline packages/packages.yaml.tmpl:58]: 1. removed `binlogctl`.
2. update etcdctl version to v3.5.15.

Comment (wuhuizuo) [reply packages/packages.yaml.tmpl:58]: /cc @Benjamin2037 @lhy1024
PTAL.

File: packages/packages.yaml.tmpl [hunk 2]
```diff
@@ -48,16 +48,73 @@ components:
...
+                  type: http
+                  url: "https://github.com/pingcap/tiup/releases/download/v1.8.1/tiup-v1.8.1-{{ .Release.os }}-{{ .Release.arch }}.tar.gz"
+                  extract: true
+                  extract_inner_path: bin/tiup-ctl
+              - name: etcdctl
+                src:
+                  type: http
+                  url: "https://github.com/etcd-io/etcd/releases/download/v3.5.15/etcd-v3.5.15-{{ .Release.os }}-{{ .Release.arch }}{{ ternary ".zip" ".tar.gz" (eq .Release.os "darwin") }}"
+                  extract: true
+                  extract_inner_path: etcd-v3.5.15-{{ .Release.os }}-{{ .Release.arch }}/etcdctl
```
Comment (wuhuizuo) [inline packages/packages.yaml.tmpl:77]: /cc @lhy1024

PTAL, from v8.4.0 update `etcdctl` binary in TiUP `ctl` package.

**Review by lhy1024 — APPROVED** [review]
lgtm for the part of etcd

File: packages/packages.yaml.tmpl (hunk 1 above)
Comment (wuhuizuo) [reply packages/packages.yaml.tmpl:58]: /cc @Benjamin2037

PTAL for point 1.

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Review by purelind — APPROVED** [review]
/lgtm

**Pull request merged**